docs/guides/nodejs/nitric-and-pgsql.mdx

```diff
@@ -0,0 +1,206 @@
+---
+description: 'How to work with Nitric SQL Databases using PostgreSQL'
+tags:
+  - Databases & CMS
+languages:
+  - typescript
+  - javascript
+published_at: 2025-03-18
+---
+
+# Nitric SQL Databases with PostgreSQL
+
+This guide demonstrates how to use PostgreSQL with Nitric's [SQL database](/sql) feature. We will be creating a simple to-do app. The finished source can be found [here](https://github.com/nitrictech/examples/tree/main/v1/nitric-pgsql).
+
+<Note>SQL databases are currently in preview and only support PostgreSQL.</Note>
+
+## Create a new Nitric project
+
+The first step is to create a new Nitric TypeScript project using the [Nitric CLI](/reference/cli).
+
+```bash
+nitric new todo-app ts-starter
+```
+
+## Enabling SQL databases
+
+SQL databases are currently in Preview. To enable this feature in your project, add the following to your `nitric.yaml` file:
+
+```yaml title: nitric.yaml
+preview:
+  - sql-databases
+```
+
+## Add a PostgreSQL client
+
+In this guide, we'll use the [node-postgres](https://github.com/brianc/node-postgres) library, but feel free to use another client of your choice and adjust the steps accordingly.
+
+```bash
+npm install pg
+```
+
+## Create a Nitric SQL Database and Connect to it
+
+Use Nitric’s `sql` feature to create a new SQL database, then use connection string and the `pg` library to connect to it.
+
+```ts title: resources/db.ts
+import { sql } from '@nitric/sdk'
+import { Client } from 'pg'
+
+const db = sql('todo-db', {
+  migrations: 'file://migrations/todos',
+})
+
+let client: Client
+
+const getClient = async () => {
+  if (!client) {
+    const connectionString = await db.connectionString()
+    client = new Client({ connectionString })
+    await client.connect()
+  }
+  return client
+}
+
+export default getClient
+```
+
+## Database Migrations
+
+Create a migration file to define the `todos` table.
+
+```sql title: migrations/todos/1_create_table.up.sql
+CREATE TABLE IF NOT EXISTS todos (
+    id SERIAL PRIMARY KEY,
+    text TEXT NOT NULL,
```

Comment:
There are highlighting issues due to "text" being a keyword in postgres... might be worth changing to "content" if updating all the screenshots isn't a massive pain. Otherwise, I think its fine.

Reply:
Updated screenshots and code.